### ISSUE TITLE
cannot build prow

### ISSUE THREAD

**Issue by jfelten** [kubernetes/test-infra#2315]
WORKSPACE_DIR=workspace
mkdir -p $WORKSPACE_DIR
cd $WORKSPACE_DIR
git clone https://github.com/kubernetes/test-infra.git
cd test-infra
bazel build //prow/...

gives:

bazel build //prow/...
INFO: Found 116 targets...
ERROR: <REDACTED>/workspace/test-infra/vendor/BUILD:65:1: null failed: Process exited with status 1 [sandboxed].
k8s.io/test-infra/vendor/github.com/Sirupsen/logrus/terminal_linux.go:10: undefined: syscall.TCGETS
Use --strategy=GoCompile=standalone to disable sandboxing for the failing actions.
INFO: Elapsed time: 1.016s, Critical Path: 0.77s

**Comment by jfelten** [comment]
$bazel version
Build label: 0.4.5-homebrew
Build target: bazel-out/local-opt/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
Build time: Thu Mar 16 13:39:33 2017 (1489671573)
Build timestamp: 1489671573
Build timestamp as int: 1489671573

$ go version
go version go1.7.3 darwin/amd64

**Comment by fejta** [comment]
What platform are you on? Also will you try cloning the repo to $GOPATH/src/k8s.io/test-infra?

**Comment by ixdy** [comment]
Looks like @jfelten is using darwin (i.e. MacOS), and we're trying to link against linux-specific code. This is https://github.com/bazelbuild/rules_go/issues/70 basically.

**Comment by fejta** [comment]
Thanks

/assign @spxtr 

It looks like we need something to deal with this: https://github.com/kubernetes/test-infra/blob/master/vendor/BUILD#L77

**Comment by fejta** [comment]
@jfelten I suspect as a workaround if you replace `terminal_linux.go` with `terminatel_bsd.go` in the above build rule it will unblock you.

Will let Joe figure out a more long term solution.

/unassign

**Comment by ixdy** [comment]
If you're trying to build prow to run on a linux kubernetes cluster you can't use bazel right now.

**Comment by fejta** [comment]
`make -C prow build` works

**Comment by jfelten** [comment]
Yes I was trying to build linux binaries on mac os., and it thought I was trying to build native binaries only.  I had set the ENV: CGO_ENABLED=0 GOOS=linux, but get the same error regardless.  I was trying to build a lightweight container with binaries only per: https://blog.codeship.com/building-minimal-docker-containers-for-go-applications/

This was my first time using bazel so please pardon my ignorance.  It probably would work if I set up a proper GOPATH and did a go get on all the sub-directories.  "make -C prow build" complains about missing go libraries.

I see that there are already docker images and kubernetes configs for each product, and will work with those instead of rolling my own.  I am converting those to a helm chart that I would contribute back if there is interest.

Should I leave this open?

**Comment by ixdy** [comment]
The issue is that Bazel really only supports native compilation, and our tooling to automanage BUILD files really [only supports `linux/amd64`](https://github.com/mikedanese/gazel/blob/a2c785aba4d9bde5e46ace135049a5607e6ea439/gazel/gazel.go#L740-L741).

This is something we want to improve, though we're blocked by Bazel and the go rules supporting cross-compilation correctly.

tl;dr we only support building with Bazel on linux/amd64 hosts right now. @fejta's suggestion in https://github.com/kubernetes/test-infra/issues/2315#issuecomment-288253261 should allow you to build the binaries on macOS, though.

**Comment by fejta** [comment]
Would definitely be interested in seeing a prow helm chart!

No need to leave open the issue unless you want spxtr to comment on it next week.

**Comment by jfelten** [comment]
I'll leave this up in case anyone else has the same problem.  I was able to work around the issue using dep:

export GOPATH=`pwd`
WORKSPACE_DIR=src
rm -rf $WORKSPACE_DIR
mkdir -p $WORKSPACE_DIR
cd $WORKSPACE_DIR
git clone https://github.com/kubernetes/test-infra.git
cd test-infra/prow/cmd/hook
go get github.com/golang/dep
go install github.com/golang/dep/cmd/dep
$GOPATH/bin/dep init
$GOPATH/bin/dep ensure
go get #just in case
CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -o hook

I should have the helm chart done later this week, but had to build my own images since it appears you guys use a private registry.

**Comment by spxtr** [comment]
I'm going to close because it's a general bazel issue, not specific to prow.
